include/boost/algorithm/is_clamped.hpp

```diff
@@ -0,0 +1,73 @@
+/* 
+   Copyright (c) Marshall Clow 2008-2012.
+
+   Distributed under the Boost Software License, Version 1.0. (See accompanying
+   file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+  
+*/
+
+/// \file is_clamped.hpp
+/// \brief IsClamped algorithm
+/// \author TODO
+///
+
+#ifndef BOOST_ALGORITHM_IS_CLAMPED_HPP
+#define BOOST_ALGORITHM_IS_CLAMPED_HPP
+
+#include <functional>       //  For std::less
+#include <iterator>         //  For std::iterator_traits
+#include <cassert>
+
+#include <boost/mpl/identity.hpp>      // for identity
```

Comment:
Does this introduce a new dependency on mp11 or was the library already using it? And does Boost.Algorithm already use Boost.mpl?

Reply:
It was already using it, from what I can find out it is a cool trick to force second and third argument to match the first one. 

https://stackoverflow.com/questions/26885239/whats-the-point-of-boostmplidentityttype-here